src/EntityFramework.Relational/RelationalDbSetExtensions.cs

```
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using JetBrains.Annotations;
+using Microsoft.Data.Entity.Infrastructure;
+using Microsoft.Data.Entity.Query;
+using Microsoft.Data.Entity.Relational;
+using Microsoft.Data.Entity.Utilities;
+using Microsoft.Framework.DependencyInjection;
+
+// ReSharper disable once CheckNamespace
+
+namespace Microsoft.Data.Entity
+{
+    public static class RelationalDbSetExtensions
+    {
+        public static IQueryable<TEntity> FromSql<TEntity>([NotNull]this DbSet<TEntity> dbSet, [NotNull]string query) where TEntity : class
+        {
+            Check.NotNull(dbSet, nameof(dbSet));
+            Check.NotNull(query, nameof(query));
+
+            if (dbSet.Context.Database as RelationalDatabase == null)
```

Without this, someone can include relational and make a call on a non-relational context (i.e. InMemoryStore) and it will silently ignore the custom SQL.